### PR TITLE
Unit tests for Vulnerability Detector

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
|[4956](https://github.com/wazuh/wazuh/issues/4956)|

## Description

This pull request adds the necessary unit tests cases to get 100% of coverage in the next functions:
**wm_vuln_detector_evr.c**
- int order();
- int pkg_version_compare();
- bool pkg_version_relate();

**wm_vuln_detector.c**
- wm_checks_package_vulnerability()
- wm_vuldet_build_url()
- wm_vuldet_compare_pkg()
- wm_vuldet_add_cve_node()
- wm_vuldet_free_cve_node()
- wm_vuldet_get_cvss()
- wm_vuldet_build_nvd_report_condition()
- wm_vuldet_fill_report_nvd_cve_info()
- wm_vuldet_fill_report_nvd_references()
- wm_vuldet_fill_report_nvd_scoring()
- wm_vuldet_fill_report_oval_data()
- wm_vuldet_send_cve_report()
- wm_vuldet_linux_oval_vulnerabilities()
- wm_vuldet_extract_advisories()
- wm_vuldet_generate_os_and_kernel_package()
- wm_vuldet_get_oslinux_info()
- wm_vuldet_discard_kernel_package()

**wm_vuln_detector_nvd.c**
- wm_vuldet_clean_version()
- wm_vuldet_linux_nvd_vulnerabilities()
- wm_vuldet_check_generic_package()
- wm_vuldet_check_specific_package()
- wm_vuldet_get_children()
- wm_vuldet_get_siblings()

**string_op.c**
- w_tolower_str()

The following functions are not completed until the PR is merge: https://github.com/wazuh/wazuh/pull/5035
- wm_vuldet_linux_rm_false_positivies()
- wm_vuldet_send_agent_report()

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** a08c803d44938d8460fdf7754139baeee895b0a2
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2084/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2084/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2084/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B2084.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2084/Analysis_dynamic_valgrind-v3.13.0-B2084.log)

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -617,14 +617,17 @@ int wm_vuldet_compare_pkg(cve_vuln_pkg *Pkg1, cve_vuln_pkg *Pkg2) {
         return -1;
     }
 
-    if (strcmp(Pkg1->version, Pkg2->version)) { // Version
+    if (!(!Pkg1->version && !Pkg2->version) &&
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:620]: @0xN3utr0n could you confirm that the addition of this conditions again is correct? The removal of those conditions was causing some failures in the unit tests and also could result in some crashes.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c [hunk 2]
```diff
@@ -1695,6 +1697,7 @@ int wm_vuldet_linux_nvd_vulnerabilities(sqlite3 *db, agent_software *agent, OSHa
     os_free(pkg_source);
     os_free(pkg_name);
     os_free(pkg_version);
+    os_free(pkg_source_version);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c:1700]: This was causing a little memory leak in some scenarios.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 359107758461ed6715505dd6a0f6bd0196cd6060
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2098/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2098/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2098/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B2098.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2098/Analysis_dynamic_valgrind-v3.13.0-B2098.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 359107758461ed6715505dd6a0f6bd0196cd6060
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2099/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2099/ossec.log)
**Coverage:** 

```json
Lines executed:26.90% of 3190

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2099/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B2099.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B2099/Analysis_dynamic_valgrind-v3.13.0-B2099.log)

**Pull request merged**